Version display

evansiroky opened the pull request:
Add an about section to the settings Popover that displays info about the current version of the UI and Server.  This PR depends on https://github.com/catalogueglobal/datatools-server/pull/115.  Also, this PR is branched off of #242.

evansiroky committed:
feat(about): add about app section that includes version info

landonreed commented:
Once the changes to #242 are approved and merged, let's merge dev into this to make it clear what the actual changes are.

evansiroky committed:
Merge branch 'dev' into version-display

landonreed commented on lib/common/components/Sidebar.js:
```diff
@@ -125,43 +143,80 @@ export default class Sidebar extends React.Component<Props, State> {
+                <td>
+                  {commit && repoUrl
+                    ? <a href={`${repoUrl}/commit/${commit}`}>{commit.slice(0, 6)}</a>
+                    : 'unkonwn'
```
Misspelling. Also, could you take this opportunity to move this (and the other messages) to draw from the i18n directory?

landonreed commented on lib/common/components/Sidebar.js:
```diff
@@ -125,43 +143,80 @@ export default class Sidebar extends React.Component<Props, State> {
+                <td>
+                  {buildTimestamp
+                    ? moment(buildTimestamp).format('LLL')
+                    : 'unkown'
```
Misspelling.

landonreed commented on lib/common/components/Sidebar.js:
```diff
@@ -184,3 +239,8 @@ class Brand extends React.Component<{expanded: boolean}> {
+// we can't destructure because that won't work with the envify plugin
+const buildTimestamp = process.env.BUILD_TIMESTAMP
+const commit = process.env.COMMIT_SHA
+const repoUrl = process.env.REPO_URL
```
These should be added to `env.yml.tmp`. Oh, never mind. They're auto-generated during `mastarm build`. Maybe change the comment a bit to describe where they come from.

landonreed commented on lib/common/components/Sidebar.js:
```diff
@@ -125,43 +143,80 @@ export default class Sidebar extends React.Component<Props, State> {
               onChange={this._toggleTutorial}>
               Hide editor tutorial
             </Checkbox>
+            <h5>About this app</h5>
```
What do you think about making the text in the "about" section smaller?

E.g.:
![image](https://user-images.githubusercontent.com/2370911/45636119-d22df500-ba74-11e8-8633-a0ea6d9964ea.png)

landonreed commented on lib/common/components/SidebarPopover.js:
```diff
@@ -36,6 +38,7 @@ export default class SidebarPopover extends React.Component<Props, State> {
 
   componentWillMount () {
     this._onResize()
+    this.props.onMount && this.props.onMount()
```
I think in other places we call this `onComponentMount`.  Not a huge deal, but we should probably choose one and be consistent.

evansiroky replied:
Taking this further, we should actually be moving all mounting logic into the constructor of a component because [as of React 16.4 componentWillMount is deprecated and in React 17 componentWillMount will be removed entirely](https://medium.com/@sgobinda007/playing-with-component-lifecycle-methods-of-react-16-3-4c946c1e6982).  This also has the added benefit of allowing us to make much cleaner containers where the mapDispatchToProps would then be an object instead of the verbose function mapping it currently is in a lot of places.  I created issue https://github.com/catalogueglobal/datatools-ui/issues/246 to keep track of this.

landonreed commented on lib/manager/actions/status.js:
```diff
@@ -200,3 +179,58 @@ export function handleFinishedJob (job: ServerJob) {
   }
 }
+
+export function handleJobResponse (response: Response, message: string) {
```
There's a lot of turbulence in this file... is this just due to alphabetization?  If so, I don't think we should make a habit of alphabetizing en masse like this as part of a separate PR.  Alphabetizing js objects is not too offensive to me, but I think moving around hundreds of lines should be isolated into a sorting-specific PR.

evansiroky replied:
Will do in https://github.com/catalogueglobal/datatools-ui/issues/245

landonreed reviewed (changes requested):
Looks good, thanks for this!  Just a couple of small items to address.  No need to remove alphabetization in the status actions file, but as a rule we should avoid doing large-scale alphabetization in feature-specific PRs (it just makes it hard to tell what's going on).

evansiroky committed:
test(messages): add message linting

evansiroky committed:
fix(messages): add some missing messages

evansiroky committed:
Merge branch 'dev' into messages-refactor

evansiroky committed:
Revert "Revert "refactor(messages): refactor the way component messages are handled""

This reverts commit 0b54d626b2a0ebbb695600f1a53cee3d407f820c.

evansiroky committed:
refactor(lint-messages): fix comment about linting for language message differences

evansiroky committed:
Merge branch 'messages-refactor' into version-display

evansiroky committed:
refactor(sidebar): change some things to address PR review comments

evansiroky commented:
This PR is now built on top of #247

evansiroky committed:
refactor(Sidebar): rename serverInfo to appInfo

To address requested change in https://github.com/catalogueglobal/datatools-server/pull/115#issuecomment-422959910
